internal/util/grpcclient/client.go

```diff
@@ -183,8 +183,14 @@ func (c *ClientBase) callOnce(ctx context.Context, caller func(client interface{
 	if err2 == nil {
 		return ret, nil
 	}
-	if err2 == context.Canceled || err2 == context.DeadlineExceeded {
-		return nil, err2
+
+	// status.Error(codes.Canceled, context.Canceled.Error()
+	// if err2 == context.Canceled || err2 == context.DeadlineExceeded {
+	// 	return nil, err2
+	// }
+
+	if !funcutil.CheckCtxValid(ctx) {
+		return ret, err2
```

Comment:
what ret is in this case? nil?

Reply:
ret doesn't matter here. we can also choose to `return nil, err2`.

Reply:
ok, please return nil, err2 to avoid confusion.